lxd/storage_volumes.go

```diff
@@ -321,6 +571,8 @@ func storagePoolVolumesGet(d *Daemon, r *http.Request) response.Response {
 		return response.SmartError(err)
 	}
 
+	allPools := poolName == ""
```

Comment:
Could you add a comment above this line to mention that this is for `/1.0/storage-volumes` as it is a little confusing in the context of `/1.0/storage-pools/{poolName}/volumes/{type}/{name}`.

Alternatively, it may be clearer to use `mux.CurrentRoute(r).GetName()` and check against the name of the handler you have created above.